### DISCUSSION TITLE
Datadog container autodiscovery

### DISCUSSION THREAD

**Discussion by jubingc** [milvus-io/milvus#37004] [Q&A and General discussion]
We want to try datadog container autodiscovery to scrape custom metrics from milvus. However, the datadog container autodiscovery needs annotations attached to the pod. Currently, helm deployment does not support it.

Alternatively, we use the datadog agent to scrape the prometheus metrics endpoint and use prometheus to scrape metrics from milvus. However, with this approach, all original labels/tags of metrics are missing in datadog. I am not sure if I configured anything wrong.

Can I know the proper way to ship milvus custom metrics to datadog? Thanks!

**Comment by yhmo** [comment]
Wait @LoveEachDay to comment.

**Reply by LoveEachDay** [reply]
@jubingc You can set up labels and annotations for milvus components using milvus-helm chart, check out the following configuration:
```
# Global labels and annotations
# If set, this will apply to all milvus components
labels: {}
annotations: {}

```

**Reply by jubingc** [reply]
@LoveEachDay Thanks. If I understand those labels and annotations correctly, they apply to all pods. It means that I cannot have custom annotations for each pod. According to the datadog container discovery, we need to specify the container name in the annotation, for example:

```
      annotations:
        ad.datadoghq.com/mixcoord.checks: |
          {
            "prometheus": {
              "instances": [
                {
                  "prometheus_endpoint": "http://%%host%%:9091/metrics",
                  "namespace": "milvus",
                  "metrics": ["*"]
                }
              ]
            }
          }
```

The container name `mixcoord` must be part of the annotation key. I am not sure how I can achieve this with values in helm.

**Reply by jubingc** [reply]
@LoveEachDay In addition, the `mixcoord` pod helm template [does not support custom annotations](https://github.com/zilliztech/milvus-helm/blob/master/charts/milvus/templates/mixcoord-deployment.yaml#L37-L42). `.Values.annotations` are attached to the [`Deployment` object](https://github.com/zilliztech/milvus-helm/blob/master/charts/milvus/templates/mixcoord-deployment.yaml#L11-L12).
